logbook-core/src/main/java/org/zalando/logbook/core/TruncatingBodyFilter.java

```diff
@@ -0,0 +1,23 @@
+package org.zalando.logbook.core;
+
+import lombok.experimental.Delegate;
+import org.apiguardian.api.API;
+import org.zalando.logbook.BodyFilter;
+
+import static org.apiguardian.api.API.Status.INTERNAL;
+import static org.zalando.logbook.core.BodyFilters.truncate;
+
+@API(status = INTERNAL)
+public final class TruncatingBodyFilter implements BodyFilter {
+
+    @Delegate
+    private final BodyFilter delegate;
+
+    public TruncatingBodyFilter(final int maxBodySize) {
+        if (maxBodySize < 0) {
+            this.delegate = (contentType, body) -> body;
```

Comment:
`-1` is the default value. That's when we don't want any truncation at all. Another way to do it would be to have this check in the bean definition itself. I can move it there.